### PR TITLE
Dockerfile

### PR DESCRIPTION
**Pull request by WillNilges**
In response to https://github.com/ComputerScienceHouse/imaginerit2022-embedded/pull/1

### CODE REVIEW

File: HACKING/Dockerfile [hunk 1]
```diff
@@ -2,14 +2,7 @@ FROM docker.io/debian
...
-# Run with:
-# podman run -dit --device=/dev/ttyUSB0:/dev/ttyUSB0:rwm --group-add keep-groups --annotation io.crun.keep_original_groups=1  14f167d6ad5e
-
-# Once you're inside, use these commands to run the demo:
-# . $HOME/esp/esp-idf/export.sh
-# cd ~/esp/esp-idf/examples/bluetooth/bluedroid/classic_bt/bt_discovery
-# idf.py -p /dev/ttyUSB0 flash monitor
+WORKDIR /root/
+RUN mkdir -p esp && cd esp && git clone --recursive https://github.com/espressif/esp-idf.git -b v4.3.2
+RUN cd esp-idf && ./install.sh esp32
```
Comment (mxmeinhold) [inline HACKING/Dockerfile]: This directory doesn't exist. Nor does install.sh. Did you mean for the workdir to be `/root/esp`?

File: HACKING/Dockerfile [hunk 2]
```diff
@@ -2,14 +2,7 @@ FROM docker.io/debian
...
+RUN cat '. $HOME/esp/esp-idf/export.sh' >> .bashrc
```
Comment (mxmeinhold) [inline HACKING/Dockerfile]: Shouldn't this be `/root/esp/...`?

File: HACKING/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ Run with:
+ ```
+podman build . --tag esp-idf
+podman run -dit --device=/dev/ttyUSB0:/dev/ttyUSB0:rwm --group-add keep-groups --annotation io.crun.keep_original_groups=1  14f167d6ad5e
+ ```
+
+Once you're inside, use these commands to run the demo:
+```
+. $HOME/esp/esp-idf/export.sh
+cd ~/esp/esp-idf/examples/bluetooth/bluedroid/classic_bt/bt_discovery
```
Comment (mxmeinhold) [inline HACKING/README.md]: I'd make these absolute too.

File: HACKING/Dockerfile [hunk 4]
```diff
@@ -2,14 +2,7 @@ FROM docker.io/debian
...
-#cat '. $HOME/esp/esp-idf/export.sh' >> ~/.bashrc
-
-# Run with:
-# podman run -dit --device=/dev/ttyUSB0:/dev/ttyUSB0:rwm --group-add keep-groups --annotation io.crun.keep_original_groups=1  14f167d6ad5e
-
-# Once you're inside, use these commands to run the demo:
-# . $HOME/esp/esp-idf/export.sh
-# cd ~/esp/esp-idf/examples/bluetooth/bluedroid/classic_bt/bt_discovery
-# idf.py -p /dev/ttyUSB0 flash monitor
+WORKDIR /root/
```
Comment (mxmeinhold) [inline HACKING/Dockerfile]: Are you presuming the user will always be root?

File: HACKING/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+idf.py -p /dev/ttyUSB0 flash monitor
+```
+
+See [the documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/index.html) for more information.
```
Comment (mxmeinhold) [inline HACKING/README.md]: nit: I'd specify `the esp-idf documentation`

File: HACKING/README.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# Contributing to this project
+
+First of all, thanks! Second of all, this is pretty much just a stub right now, but here's how you use the Dockerfile.
+
+## Dockerfile
+ Run with:
+ ```
+podman build . --tag esp-idf
```
Comment (mxmeinhold) [inline HACKING/README.md]: Might be good to link to podman install docs?

File: HACKING/Dockerfile (hunk 4 above)
Comment (WillNilges) [reply HACKING/Dockerfile]: Yep, I'm running this in an unprivileged Podman container, so I presume root is fine. I should make it clear that this Dockerfile is supposed to be used for development.

File: HACKING/README.md (hunk 6 above)
Comment (WillNilges) [reply HACKING/README.md]: I feel like that's out of scope, but It's no trouble to add them.

Comment (mxmeinhold) [reply HACKING/README.md]: imo if you're inviting people to help, it's worth giving more guidance, especially as podman is not as broadly known as other tools.

File: HACKING/Dockerfile [hunk 7]
```diff
@@ -1,15 +1,9 @@
...
-# podman run -dit --device=/dev/ttyUSB0:/dev/ttyUSB0:rwm --group-add keep-groups --annotation io.crun.keep_original_groups=1  14f167d6ad5e
-
-# Once you're inside, use these commands to run the demo:
-# . $HOME/esp/esp-idf/export.sh
-# cd ~/esp/esp-idf/examples/bluetooth/bluedroid/classic_bt/bt_discovery
-# idf.py -p /dev/ttyUSB0 flash monitor
+RUN apt-get -y update && apt-get -y install flex bison gperf python3 python3-pip python3-setuptools cmake ninja-build ccache libffi-dev libssl-dev dfu-util libusb-1.0-0 unzip
+RUN apt-get -y install vim git wget
+WORKDIR /root/
+RUN mkdir -p esp && cd esp && wget https://github.com/espressif/esp-idf/archive/refs/tags/v4.3.2.zip -O esp-idf.zip && unzip esp-idf.zip
```
Comment (mxmeinhold) [inline HACKING/Dockerfile]: nit: Why not `WORKDIR /root/esp`?

**Review by mxmeinhold — APPROVED** [review]
LGTM! :ship: :it:

**Pull request merged**